ska_trend/fid_drop_mon/update.py

```diff
@@ -0,0 +1,384 @@
+import argparse
+import functools
+from pathlib import Path
+
+import astropy.units as u
+import jinja2
+import numpy as np
+from acdc.common import send_mail
+from astropy.table import Table, join, vstack
+from cheta import fetch
+from cheta.utils import logical_intervals
+from cxotime import CxoTime, CxoTimeLike
+from kadi import events
+from kadi.commands.observations import get_observations, get_starcats
+from ska_helpers.logging import basic_logger
+from ska_helpers.run_info import log_run_info
+
+from ska_trend import __version__
+
+DOC_ID = "1GoYBTIQAv0qq2vh3jYxHBYHfEq2I8LVGMiScDX7OFvw"
+GID = "1266351479"
+url_start = "https://docs.google.com/spreadsheets/d"
+GSHEET_URL = f"{url_start}/{DOC_ID}/export?format=csv&id={DOC_ID}&gid={GID}"
+GSHEET_USER_URL = f"{url_start}/{DOC_ID}/edit?usp=sharing"
+
+# Constants and file path definitions
+FILE_DIR = Path(__file__).parent
+
+
+def INDEX_TEMPLATE_PATH():
+    return FILE_DIR / "index_template.html"
+
+
+URL = "https://cxc.cfa.harvard.edu/mta/ASPECT/fid_drop_mon/index.html"
+LOGGER = basic_logger(__name__, level="INFO")
+
+fetch.data_source.set("cxc")
+
+
+def get_opt():
+    parser = argparse.ArgumentParser(description="Fid drop monitor")
+    parser.add_argument("--start", help="Start date")
+    parser.add_argument("--stop", help="Stop date")
+    parser.add_argument(
+        "--out-dir",
+        default="./out",
+        help="Output directory",
+    )
+    parser.add_argument(
+        "--email",
+        action="append",
+        dest="emails",
+        default=[],
+        help="Email address for notification",
+    )
+    return parser
+
+
+def get_scs107_intervals(start: CxoTimeLike, stop: CxoTimeLike) -> Table:
```

Comment:
And @taldcroft this is where I used a little helper function I found useful to get the times when we were running just the vehicle loads until the next cmd from new products.  The function doesn't really belong in this fid_drop code, it isn't well tested, and it is a little hacky to just have the other scs107 times in here, but something like this could be useful for a quick review in other code when we want to find when planned obsids are different from telemetered obsids.

Reply:
Yes, this seems useful so I spent some time reviewing it carefully. We can probably leave it here for this PR and then perhaps move it somewhere else later. At some level it feels like a kadi commands utility.

I think it should be named `get_vehicle_only_intervals`. An "SCS-107 interval" is pretty ambiguous.

Reply:
I'm tempted to move the function into kadi sooner rather than later just to have some tests.